### PR TITLE
Add UserDefaults improvements - search, editing, value on the list screen

### PR DESCRIPTION
**Pull request by ZvonimirMedak**
## Summary

I have added a search functionality to the SentinelListView which is also used for the UserDefaults, added

### Type

- [X] **Feature**: This pull request introduces a new feature.
- [ ] **Bug fix**: This pull request fixes a bug.
- [ ] **Refactor**: This pull request refactors existing code.
- [ ] **Documentation**: This pull request updates documentation.
- [ ] **Other**: This pull request makes other changes.

#### Additional information

- [ ] This pull request introduces a **breaking change**.

### Description

I have added searchableIfAvaialble modifier since searchable is only available from iOS 15.
I have added the functionality to edit some user default properties like String, Bool, Data (base64Encoded), Double, Float, Int, and [String].
I have added the value of the user defaults property below the name on the screen where the user defaults properties are listed

## Checklist

- [X] I have performed a self-review of my own code.
- [X] I have tested my changes, including edge cases.
- [X] I have added necessary tests for the changes introduced (if applicable).
- [X] I have updated the documentation to reflect my changes (if applicable).

## Additional notes

https://github.com/user-attachments/assets/09792ff5-4b0a-4d86-8b6d-ed4468a1c405
https://github.com/user-attachments/assets/a1a332a9-0c6b-4c26-af23-24a23e281ca1

### CODE REVIEW

File: Sentinel/Classes/Core/Internal/SentinelListViewModel.swift [hunk 1]
```diff
@@ -0,0 +1,46 @@
...
+
+import Combine
+
+final class SentinelListViewModel: ObservableObject {
+
+    // MARK: - Internal properties
+
+    @Published var serachText: String = ""
+    @Published var sections: [ToolTableSection]
+    let initialSections: [ToolTableSection]
```
Comment (nikolamajcen) [inline Sentinel/Classes/Core/Internal/SentinelListViewModel.swift:16]: From the calling side you will see `initialSections` which may create a confusion - what type of sections there could also be. I would rather use `sections` if needed, and internally we can do different kind of "magic" with those.

Or, maybe better, update the `init` method with proper parameter name, but in that case, the property `initialSections` should not be exposed due to potential confusion (it is not used outside of the object).

In short;
* Make `initialSections` private
* Change the parameter name in the `init` method to `sections`

**Review by nikolamajcen — CHANGES_REQUESTED** [review]

**Review by nikolamajcen — APPROVED** [review]
